File: sdkwork-backend-sdk-java/src/main/java/com/sdkwork/backend/api/CharacterGenerationApi.java
package com.sdkwork.backend.api;

import com.sdkwork.backend.http.HttpClient;
import com.sdkwork.backend.model.*;
import java.util.List;
import java.util.Map;

public class CharacterGenerationApi {
    private final HttpClient client;
    
    public CharacterGenerationApi(HttpClient client) {
        this.client = client;
    }

    /** Create character generation task */
    public PlusApiResultGenerateCharacterVO create(GenerateCharacterForm body) throws Exception {
        return (PlusApiResultGenerateCharacterVO) client.post(ApiPaths.backendPath("/generation/character/create"), body);
    }

    /** Get character generation result */
    public PlusApiResultGenerateCharacterVO getResult(String taskId) throws Exception {
        return (PlusApiResultGenerateCharacterVO) client.get(ApiPaths.backendPath("/generation/character/result/" + taskId + ""));
    }
}

File: sdkwork-backend-sdk-swift/Sources/API/CharacterGenerationApi.swift
import Foundation

public class CharacterGenerationApi {
    private let client: HttpClient
    
    public init(client: HttpClient) {
        self.client = client
    }

    /// Create character generation task
    public func create(body: GenerateCharacterForm) async throws -> PlusApiResultGenerateCharacterVO? {
        let response = try await client.post(ApiPaths.backendPath("/generation/character/create"), body: body)
        return response as? PlusApiResultGenerateCharacterVO
    }

    /// Get character generation result
    public func getResult(taskId: String) async throws -> PlusApiResultGenerateCharacterVO? {
        let response = try await client.get(ApiPaths.backendPath("/generation/character/result/\(taskId)"))
        return response as? PlusApiResultGenerateCharacterVO
    }
}
